frontend/src/container/FormAlertRules/QuerySection.tsx

```diff
@@ -258,11 +283,41 @@ function QuerySection({
 			</div>
 		);
 	};
-	return (
-		<>
-			<StepHeading> {t('alert_form_step1')}</StepHeading>
-			<FormContainer>
-				<div style={{ display: 'flex' }}>
+
+	const handleRunQuery = (): void => {
+		runQuery();
+	};
+
+	const renderTabs = (typ: AlertTypes): JSX.Element | null => {
```

Comment:
can we make a separate component in more readable format

Reply:
renderTabs() has quite a few dependencies on QuerySection. So will not be able to move this in current time, but the we are going to add query builder tab soon to logs. will move it at that time. hope thats fine